lib/std/fmt.zig

```diff
@@ -1125,6 +1125,90 @@ pub fn formatIntBuf(out_buf: []u8, value: anytype, base: u8, uppercase: bool, op
     return fbs.pos;
 }
 
+/// Formats a duration in nanoseconds according to its magnitude:
+///
+/// - #ns
+/// - #[.#]us
+/// - #[.#]ms
+/// - #[.#]s
+/// - #m#[.#]s
+/// - #h#m#[.#]s
+/// - #d#h#m#[.#]s
+/// - #w#d#h#m#[.#]s
+pub fn formatDuration(ns: u64, writer: anytype) !void {
+    var ns_remaining = ns;
+
+    inline for (.{
+        .{ .ns = std.time.ns_per_week, .sep = 'w' },
+        .{ .ns = std.time.ns_per_day, .sep = 'd' },
+        .{ .ns = std.time.ns_per_hour, .sep = 'h' },
+        .{ .ns = std.time.ns_per_min, .sep = 'm' },
+    }) |unit| {
+        // Compare to total to retain intermediate units
+        if (ns >= unit.ns) {
+            const units = ns_remaining / unit.ns;
+            ns_remaining -= units * unit.ns;
+            try format(writer, "{}{c}", .{ units, unit.sep });
```

Comment:
Calling `format` over and over incurs in a compile-time penalty as the format string is "exploded" every time into its basic parts.
Consider using `writer.writeByte`/`writer.writeAll`/`fmt.formatInt` directly.

Reply:
Can do!